server.go

```diff
@@ -336,3 +342,17 @@ func (sb *SplitBytes) Next() bool {
 func (sb *SplitBytes) Chunk() []byte {
 	return sb.currentChunk
 }
+
+func (s *Server) registerPlugin(p plugin) {
+	s.pluginMtx.Lock()
+	s.plugins = append(s.plugins, p)
+	s.pluginMtx.Unlock()
```

Comment:
It was more out of the concern that registration (and initialization) of multiple plugins could take long enough that the first flush happens before the plugin registration is complete. We can do away with the mutex though as long as we stick with the requirement that plugins are loaded synchronously and in series.